src/fidesops/service/connectors/saas_connector.py

```diff
@@ -310,3 +338,90 @@ def _unwrap_response_data(saas_request: SaaSRequest, response: Response) -> Any:
             raise FidesopsException(
                 f"Unable to parse JSON response from {saas_request.path}"
             )
+
+    @staticmethod
+    def _invoke_read_request_override(
+        override_function_name: str,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        node: TraversalNode,
+        input_data: Dict[str, List],
+        secrets: Any,
+    ) -> List[Row]:
+        """
+        Invokes the appropriate user-defined SaaS request override for read requests.
+
+        Contains error handling for uncaught exceptions coming out of the override.
+        """
+        override_function: Callable[
+            ..., Union[List[Row], int]
+        ] = SaaSRequestOverrideFactory.get_override(
+            override_function_name, SaaSRequestType.READ
+        )
+        try:
+            return override_function(
+                node,
+                policy,
+                privacy_request,
+                input_data,
+                privacy_request.get_cached_identity_data(),
+                secrets,
+            )  # type: ignore
+        except Exception:
+            logger.error(
+                "Encountered error executing override access function '%s",
+                override_function_name,
+                exc_info=True,
+            )
+            raise FidesopsException(
+                f"Error executing override access function '{override_function_name}'"
+            )
+
+    @staticmethod
+    def _invoke_masking_request_override(
+        override_function_name: str,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        rows: List[Row],
+        query_config: SaaSQueryConfig,
+        masking_request: SaaSRequest,
+        secrets: Any,
+    ) -> int:
+        """
+        Invokes the appropriate user-defined SaaS request override for masking
+        (update, delete, data_protection_request) requests.
+
+        Includes the necessary data preparations for override input
+        and has error handling for uncaught exceptions coming out of the override
+        """
+        override_function: Callable[
+            ..., Union[List[Row], int]
+        ] = SaaSRequestOverrideFactory.get_override(
+            override_function_name, SaaSRequestType(query_config.action)
+        )
+        try:
+            # if using a saas override, we still need to use the core framework
+            # to generate updated (masked) parameter values, and pass these
+            # into the overriden function
+            update_param_values: List[Dict[str, Any]] = [
+                query_config.generate_update_param_values(
+                    row, policy, privacy_request, masking_request
+                )
+                for row in rows
+            ]
+            return override_function(
+                update_param_values,
+                policy,
+                privacy_request,
+                privacy_request.get_cached_identity_data(),
```

Comment:
What are your thoughts on just passing `privacy_request` and getting the identity data if needed from the privacy request?

Reply:
yup, i like that - anything to reduce the number of params is better IMO